### PR TITLE
Move add to hook queue for created repo to outside xorm session.

### PR DESCRIPTION
**Pull request by davidsvantesson**
New try to solve issue #7404. Moving adding created repository to hook queue outside the xorm session. This code is therefore duplicated and moved up to functions ForkRepository and CreateRepository. This is necessary to ensure the hook is in the database when read by DeliveryHooks.

### CODE REVIEW

File: models/repo.go [hunk 1]
```diff
@@ -1402,7 +1401,14 @@ func CreateRepository(doer, u *User, opts CreateRepoOptions) (_ *Repository, err
 		}
 	}
 
-	return repo, sess.Commit()
+	err = sess.Commit()
+
+	// Add to hook queue for created repo after session commit.
+	if u.IsOrganization() {
```
Comment (lafriks) [inline models/repo.go:1409]: If commit fails repo won't be created

Comment (davidsvantesson) [reply models/repo.go:1409]: Error handling for session commit added.

I do not know why amd64 build is failing after I added this commit.

Comment (lunny) [reply models/repo.go:1409]: That's arm64, I think that's fail unrelated.

Comment (techknowlogick) [reply models/repo.go:1409]: I restarted build, it was amd64 fail first, now arm64. I restarted again, hopefully that fixes things.

File: models/repo.go [hunk 2]
```diff
@@ -1402,7 +1401,17 @@ func CreateRepository(doer, u *User, opts CreateRepoOptions) (_ *Repository, err
 		}
 	}
 
-	return repo, sess.Commit()
+	err = sess.Commit()
```
Comment (lafriks) [inline models/repo.go]:
```suggestion
	if err = sess.Commit(); err != nil {
```

**Review by lafriks — CHANGES_REQUESTED** [review]

**Review by lunny — APPROVED** [review]

File: models/repo.go (hunk 2 above)
Comment (davidsvantesson) [reply models/repo.go]: Fixed.

**Review by techknowlogick — APPROVED** [review]

**Comment by lunny** [comment]
@lafriks Please review again.

**Review by lafriks — APPROVED** [review]

**Comment by lunny** [comment]
@davidsvantesson please send back port to v1.9

**Pull request merged**